src/Compilers/Core/Portable/IReferenceOrISignature.cs

```diff
@@ -3,6 +3,7 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Diagnostics;
```

Comment:
:memo: All the circularity and self-reference tests fail in UsedAssemblies branch (ie. when emitting). Those scenarios needs to be blocked with an error. I'm working on this now.